packages/astro/src/content/types-generator.ts

```diff
@@ -46,6 +46,10 @@ type CollectionEntryMap = {
 		| {
 				type: 'data';
 				entries: Record<string, DataEntryMetadata>;
+		  }
+		| {
+				type: 'experimental_data';
+				entries: Record<string, DataEntryMetadata>;
```

Comment:
Is this how we plan to ship the feature as experimental?

Reply:
This is something we should discuss. It has the benefit of meaning it's backward-compatible, as it can be used alongside the existing collection types.